### PR TITLE
Implement new `EntityObjectLoaderRtkQuery` that uses RTK Query hooks

### LINKED ISSUE #44491: Implement new `EntityObjectLoaderRtkQuery` that uses RTK Query hooks
Make it work for 1 or 2 entities: databases and/or tables.

There is a separate issue to port the lest of entities: #50323

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #44491

Part of #50317

### Description

Adds `EntityObjectLoaderRtkQuery` which is a replacement for `EntityObjectLoader`.
See [[Tech] Migrate entity framework loaders to RTK Query](https://www.notion.so/metabase/Tech-Migrate-entity-framework-loaders-to-RTK-Query-14369354c90180ecadc2d1f1a5dba6e0) for details.
Enables the new component for 2 types of entities: databases & tables.
Other entities will be ported in separate PRs, and then the old `EntityObjectLoader` will be removed.

### How to verify

CI is green.

### CODE REVIEW

File: frontend/src/metabase/entities/containers/EntityObjectLoader.jsx [hunk 1]
```diff
@@ -142,10 +142,6 @@ class EntityObjectLoaderInner extends Component {
       },
     );
   };
-
-  remove = () => {
```
Comment (kamilmielnik) [inline frontend/src/metabase/entities/containers/EntityObjectLoader.jsx:146]: Unused

File: frontend/src/metabase/entities/containers/EntityObjectLoader.jsx [hunk 2]
```diff
@@ -146,10 +146,6 @@ class EntityObjectLoaderInner extends Component {
...
```
Comment (kamilmielnik) [inline frontend/src/metabase/entities/containers/EntityObjectLoader.jsx:150]: Unused

File: frontend/src/metabase/entities/containers/rtk-query/types.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import type { UseQuery } from "@reduxjs/toolkit/dist/query/react/buildHooks";
```
Comment (kamilmielnik) [inline frontend/src/metabase/entities/containers/rtk-query/types.ts:1]: Entity framework internals meet TypeScript for the first time.
Beware: types in this file are incomplete and may be inaccurate.

File: frontend/src/metabase/entities/containers/rtk-query/types.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    object: unknown;
+  };
+  rtk: {
+    getUseGetQuery: (fetchType: FetchType) => {
+      action?: string;
+      transformResponse?: (data: Entity, query: EntityQuery) => unknown;
+      useGetQuery: UseQuery<
+        QueryDefinition<unknown, BaseQueryFn, TagType, Entity>
+      >;
+    };
```
Comment (kamilmielnik) [inline frontend/src/metabase/entities/containers/rtk-query/types.ts:51]: `getUseListQuery` will be added here in [milestone 2](https://github.com/metabase/metabase/issues/50317)

File: frontend/src/metabase/entities/containers/rtk-query/EntityObjectLoaderRtkQuery.tsx [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+/**
+ * Replacement for EntityObjectLoader for better compatibility with RTK Query.
+ *
+ * For example, generic types for the Database entity would be:
+ *   Entity        -> Database from metabase-types/api/database.ts
+ *   EntityWrapper -> Database from metabase-lib/v1/metadata/Database.ts
+ *
+ * @deprecated use "metabase/api" instead
+ */
+export function EntityObjectLoaderRtkQuery<Entity, EntityWrapper>({
```
Comment (kamilmielnik) [inline frontend/src/metabase/entities/containers/rtk-query/EntityObjectLoaderRtkQuery.tsx:88]: The `RtkQuery` suffix is temporary and is not intended to go to `master`.
The suffix will be removed once the old `EntityObjectLoader` can be removed (#50322).

File: frontend/src/metabase/entities/tables.js [hunk 6]
```diff
@@ -51,6 +56,30 @@ const Tables = createEntity({
...
+
+      if (fetchType === "fetchMetadataDeprecated") {
+        return {
+          useGetQuery: useGetTableQueryMetadataQuery,
+        };
+      }
+
+      if (fetchType === "fetchMetadataAndForeignTables") {
+        throw new Error("Deprecated, use 'metabase/api' instead");
+      }
```
Comment (kamilmielnik) [inline frontend/src/metabase/entities/tables.js:75]: TODO

File: frontend/src/metabase/entities/containers/rtk-query/types/index.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
+import type { BaseQueryFn, QueryDefinition } from "@reduxjs/toolkit/query";
```
Comment (kamilmielnik) [inline frontend/src/metabase/entities/containers/rtk-query/types/index.ts:1]: Entity framework internals meet TypeScript for the first time.
Beware: types in this file are incomplete and may be inaccurate.

File: frontend/src/metabase/visualizations/visualizations/CartesianChart/use-chart-events.ts [hunk 8]
```diff
@@ -79,7 +85,7 @@ export const useChartEvents = (
     [chartModel.seriesModels, hovered],
   );
 
-  useEffect(
+  useLayoutEffect(
```
[the remark this answers is not in the document]
Comment (kamilmielnik) [reply frontend/src/metabase/visualizations/visualizations/CartesianChart/use-chart-events.ts:88]: I'll make sure to add a comment here before this is merged.

**Review by ranquild — APPROVED** [review]
Super!

**Review by alxnddr — APPROVED** [review]

File: frontend/src/metabase/visualizations/visualizations/CartesianChart/use-chart-events.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (kamilmielnik) [reply frontend/src/metabase/visualizations/visualizations/CartesianChart/use-chart-events.ts:95]: Added a comment: 63dfebdff26e20b4c275049df6c05bb34a3a22c9

**Pull request merged**